woocommerce/woocommerce
# Remove Purchase task from onboarding task list

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR removes the Purchase task from the default onboarding tasks lists. We no longer use this task in our current onboarding, and it has been causing thousands of API requests on some sites. 

Basically, the task lists are initialized on every wp-admin page load to determine how many unfinished items to show in the sidebar notifications badge. When the Purchase task is initialized, it hits our API to see how many addons and themes are available for purchase. The response is supposed to be saved as a transient, but if that fails for some reason (as it seems to be on some sites), it will make multiple API requests for each wp-admin page load.

Closes #39955.



### How to test the changes in this Pull Request:



### Updated testing instructions

In this testing instruction, we're confirming that the purchase task is not displayed when we've manually overridden the onboarding data with selected "Memberships" product type and a paid theme "deco". In previous versions, this is how it [would look](https://github.com/woocommerce/woocommerce/assets/3747241/3fbf581d-e6c1-465e-8b92-011707765f7d).

1. Skip onboarding wizard
1. Go to WCA Test Helper > Options
2. Search for the option `woocommerce_onboarding_profile`
3. Edit the option to the following value:

```
{
  "is_agree_marketing": false,
  "store_email": "store@example.com",
  "is_store_country_set": true,
  "industry": [
    {
      "slug": "fashion-apparel-accessories"
    }
  ],
  "product_types": [
    "memberships"
  ],
  "product_count": "0",
  "selling_venues": "no",
  "setup_client": false,
  "business_extensions": [],
  "theme": "deco",
  "completed": true
}
```

4. Go to WooCommerce > Home
5. Observe that you don't have a task with the title `Add Memberships and 1 more product to my store`
